packages/alias/test/test.js

```diff
@@ -144,79 +114,26 @@ test('Leaves entry file untouched if matches alias', (t) => {
   t.is(resolved, null);
 });
 
-test('Test for the resolve property', (t) => {
-  const result = alias({
-    resolve: ['.js', '.jsx'],
-    entries: [{ find: 'ember', replacement: './folder/hipster' }]
-  });
-
-  const resolved = result.resolveId('ember', posix.resolve(DIRNAME, './fixtures/index.js'));
-
-  t.is(resolved, posix.resolve(DIRNAME, './fixtures/folder/hipster.jsx'));
-});
-
 test('i/am/a/file', (t) => {
   const result = alias({
-    entries: [{ find: 'resolve', replacement: 'i/am/a/file' }]
+    entries: [{ find: 'resolve', replacement: 'i/am/a/file' }],
+    customResolver: x => x
   });
 
   const resolved = result.resolveId('resolve', '/src/import.js');
 
   t.is(resolved, 'i/am/a/file');
 });
 
-test('i/am/a/local/file', (t) => {
```

Comment:
Again, it would be interesting to reintroduce these as integration tests involving Rollup and possibly rollup-plugin-node-resolve. Actually I think it would be nice to run ALL tests through Rollup, this would make the tests much more realistic and we could also get rid of the `customResolve: x => x` that needed to be introduced into many tests to keep the structure. Here is a helper that would work:

```js
// "plugins" is an array of plugins for Rollup, they should include "alias"
// "tests" is an array of pairs [importee, importer]
function resolveWithRollup(plugins, tests) {
  return new Promise((resolve, reject) => {
    rollup({
      input: 'dummy-input',
      plugins: [
        {
          name: 'test-plugin',
          buildStart() {
            resolve(
              // The buildStart hook is the first to have access to this.resolve
              // We map the tests to an array of resulting ids
              Promise.all(
                tests.map(([importee, importer]) =>
                  this.resolve(importee, importer).then(({ id }) => id)
                )
              )
            );
          },
          resolveId(id) {
            if (id === 'dummy-input') return id;
            return null;
          },
          load(id) {
            if (id === 'dummy-input') return 'console.log("test");';
            return null;
          }
        },
        ...plugins
      ]
    // if Rollup throws an error, this should reject the test
    }).catch(reject);
  });
}

// convenience function for just injecting the alias plugin
function resolveAliasWithRollup(aliasOptions, tests) {
  return resolveWithRollup([alias(aliasOptions)], tests);
}
```

Here is an example test, rewritten using the helper:

```js
test('Simple aliasing (array)', (t) =>
  resolveAliasWithRollup(
    {
      entries: [
        { find: 'foo', replacement: 'bar' },
        { find: 'pony', replacement: 'paradise' },
        { find: './local', replacement: 'global' }
      ]
    },
    [
      ['foo', '/src/importer.js'],
      ['pony', '/src/importer.js'],
      ['./local', '/src/importer.js']
    ]
  ).then((result) => t.deepEqual(result, ['bar', 'paradise', 'global'])));
```

For tests involving the alias plugin and other plugins, the first helper could be used.

Reply:
Thanks for detailed example. I was thinking in same way, but stopped at dilemma: to include rollup-plugin-node-resolve or not. My point is that it is not a good idea to include 3rd party plugin into  `alias` plugin tests as it is not reliable in long-term. But right now i do not see any better option to keep tests coverage at good level without doing it.

Reply:
Probably it would be better to include rollup-plugin-node-resolve after https://github.com/rollup/plugins/pull/32 is merged. However it does not block other changes.

Reply:
Done. Please find changes in latest version.

Reply:
> My point is that it is not a good idea to include 3rd party plugin into alias plugin tests as it is not reliable in long-term

It will make the test slightly less stable, but as rollup-plugin-node-resolve is an important pillar of the Rollup ecosystem, it will basically improve consistency of the wider ecosystem, which I would love very much.

> Probably it would be better to include rollup-plugin-node-resolve after #32 is merged

That would definitely make it easier, but I am not sure how long this will take, this is not a trivial move. If you reference the external plugin for now, we should just not forget to change the reference once #32 is merged. Should not be a big change, but this does not need to be coupled.